### ISSUE TITLE
Is there a check for consistent spacing for newline escapes?

### ISSUE THREAD

**Issue by rovinski** [nmoroze/tclint#17]
Here is an example which is not flagged as of 0.2.0 with default options:
```
set var {one \
         two\
         three  \
         four\
         five   \
         six \
         seven
}
```
Is there a check for this? I think by default I would expect a single space:
```
set var {one \
         two \
         three \
         four \
         five \
         six \
         seven
}
```
Or maybe a user could override with the number of spaces (e.g. 0):
```
set var {one\
         two\
         three\
         four\
         five\
         six\
         seven
}
```
And in an advanced case maybe it could check for block alignment:
```
set var {one   \
         two   \
         three \
         four  \
         five  \
         six   \
         seven
}
```

Thanks!

**Comment by nmoroze** [comment]
This check doesn't exist yet, but I think it's a good suggestion - thanks! I'll go ahead and implement it and close this issue once it's done. 

I also agree that a single space would be my baseline expectation. I'll probably start with that to minimize proliferation of options, but open to adding more configuration flexibility if you think alternative styles are fairly common.

**Comment by nmoroze** [comment]
Thinking a bit more about this -`tclint` may not want to check this inside curly braces, since the spacing does technically affect the contents of the braced word. For example, these two commands have a different result:

```tcl
puts {one \
      two}
# output: one  two

puts {one\
      two}
# output: one two
```

In practice it may not matter, since I think most people default to `""` for displayed strings (the main place where I imagine subtle whitespace differences count), but I'm not sure. That being said, I also think in most cases the curly braces make the newline escapes unnecessary! It depends how commands interpret their arguments, but in the example of `sta::define_cmd_args` which I think spurred this issue, it seems like they might not be needed: https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/rcx/src/OpenRCX.tcl#L54-L64.

I guess that does beg the question of whether `tclint` could flag these unnecessary newline escapes... which unfortunately I think would be tricky given it's command-specific.

**Comment by rovinski** [comment]
Very fair... I think it could still check newline escapes outside of curly braces. Maybe there's a larger issue of flagging when to use curly braces vs. quotes? Seems difficult though.

**Comment by maliberty** [comment]
Likewise with inside double quoted strings.
